### PR TITLE
Added benefit eligibility api for sorted lists

### PR DESCRIPTION
**Pull request by DaceyTom2**
# Description

[Return sorted benefits based off of eligibility criteria](https://bf-decd.atlassian.net/browse/BF-92?atlOrigin=eyJpIjoiNzMzZGNkYzVjYjIwNGE4OWI1YmRhYjI0MzlmNWM0NjMiLCJwIjoiaiJ9)

Added benefit eligibility list api. Doing a get on this api will return a sorted list of benefits based off of the situation cookie.

## Meets Definition of Done
- [x] Unit tests up to date
- [x] E2E tests up to date
- [x] I18n

### CODE REVIEW

**Comment by DaceyTom2** [comment]
I did a first pass on this change and am looking for assistance on what is good, bad and ugly about it.

Currently to implement we need the following

- Situation cookie to be implemented
- Catalog to get benefits list from here
- Benefits in Strapi to have a unique key amongst verticals
- Additional eligibility criteria coded

**Comment by var-kyle** [comment]
There are a couple of things I would change here.

First, the file name of the api file could probably just be changed to `benefits.js` and that endpoint would just be where we get our benefits. It might just be personal preference, but I believe that `/api/benefits` is more RESTful than `/api/benefitEligibilityList`.

I don't really think creating a class to determine the eligibility is necessary here, it could easily just be another function. The class isn't really providing anything significant especially with how eligibility is being determined here. [Take a look](https://github.com/DTS-STN/c19-benefits-node/blob/master/routes/results/getBenefits.js) at how the benefits are added in the c19-benefits-finder, it may be relevant to how we get eligibility for now.

File: pages/api/benefits.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+export default async (req, res) => {
+  if (req.method === "GET") {
+    //Get situation cookie
+    let situation = {};
+    if (req.cookies.situation) {
+      situation = JSON.parse(req.cookies.situation);
+    }
+
+    //Create list of benefits and set eligibility
+    let benefits = await getBenefits("en");
```
Comment (var-kyle) [inline pages/api/benefits.js:12]: You should be able to get the language from the request so this doesn't need to be hardcoded to "en".

File: pages/api/benefits.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+const isUnder65 = (situation) => {
+  if (!situation["age"]) return eligiblityStatus.POTENTIALLY_ELIGIBLE;
+  else if (situation["age"] < 65) return eligiblityStatus.ELIGIBLE;
+  return eligiblityStatus.NOT_ELIGIBLE;
+};
+
+const eligibilityCriteria = {
+  3: [isOver60, isUnder65],
+  5: [isOver60],
+};
```
Comment (var-kyle) [inline pages/api/benefits.js:72]: These could all be moved into another file under `/lib`, but I'm ok with it for now.

**Review by var-kyle — CHANGES_REQUESTED** [review]
Looks a lot better than earlier! Just 1 small change and I'll approve it.

File: pages/api/benefits.js (hunk 1 above)
Comment (DaceyTom2) [reply pages/api/benefits.js]: Language is now from the accept-language header

File: pages/api/benefits.js (hunk 2 above)
Comment (DaceyTom2) [reply pages/api/benefits.js]: I will leave for now since the rule engine concept is still up in the air

**Pull request merged**